common/src/main/java/hardcorequesting/common/items/QuestBookItem.java

```diff
@@ -114,9 +92,10 @@ public InteractionResultHolder<ItemStack> use(Level world, Player player, @NotNu
     @Override
     public void appendHoverText(ItemStack stack, TooltipContext tooltipContext, List<Component> tooltip, TooltipFlag tooltipFlag) {
         if (enabled) {
-            CompoundTag hqmTag = stack.getOrDefault(DataComponents.CUSTOM_DATA, CustomData.EMPTY).getUnsafe().getCompound("hqm");
-            if (hqmTag.contains(NBT_PLAYER)) {
-                Player useAsPlayer = QuestingData.getPlayer(hqmTag.getString(NBT_PLAYER));
+            UUID uuid = stack.get(ModItems.DataComponents.USE_AS_PLAYER.get());
+            if (uuid != null) {
+                //FIXME QuestingData.getPlayer() will try and get the player from the mc server, but this is called from client-side!
```

Comment:
Whats up with that?

Reply:
I noticed this issue while going through the code, but I lacked any suggestion on how to change it. I would assume that the code has been like that for a good while.

Without a clear solution, I was originally thinking of just leaving it with a note, but it would be nice to solve it here and now. Looking more into it now, it would be possible to get the username from the UUID through `SkullBlockEntity.fetchGameProfile()`, but that requires meddling with concurrency, and the username still need to be stored somewhere (see `ResolvableProfile` and what player skulls are doing).
However, a much simpler solution is to just store the username serverside at op book creation into the same data component as the UUID.

Reply:
I completely trust you with the solution, I have no idea what it is about.

Reply:
Implemented said "much simpler solution". Also did some extra cleanup of the quest book item that is worth looking through.